### PR TITLE
convert: add option to symlink instead of copying

### PR DESCRIPTION
**Pull request by kierdavis**
As proposed in #2324.

### CODE REVIEW

File: beetsplug/convert.py [hunk 1]
```diff
@@ -156,6 +157,12 @@ def commands(self):
                               help=u'set the target format of the tracks')
         cmd.parser.add_option('-y', '--yes', action='store_true', dest='yes',
                               help=u'do not ask for confirmation')
+        cmd.parser.add_option('-l', '--link', action='store_true', dest='link',
+                              help=u'symlink files that do not need \
+                                     transcoding, instead of copying them; \
+                                     this may cause files in your library to \
+                                     be modified, if options such as "embed" \
+                                     are enabled')
```
Comment (sampsyo) [inline beetsplug/convert.py]: The in-line help string should usually be shorter than this—just one phrase is fine. We can leave the full detail for the documentation.

File: docs/plugins/convert.rst [hunk 2]
```diff
@@ -86,6 +93,13 @@ file. The available options are:
 - **threads**: The number of threads to use for parallel encoding.
   By default, the plugin will detect the number of processors available and use
   them all.
+- **link**: By default, files that do not need to be transcoded will be copied
+  to their destination. This option creates symbolic links instead. Note that
+  options such as ``embed`` that modify the output files after the transcoding
+  step will cause the original files to be modified as well if ``link`` is
+  enabled. For this reason, it is highly recommended not use to ``link`` and
+  ``embed`` at the same time.
+  Default: ``false``.
```
Comment (sampsyo) [inline docs/plugins/convert.rst]: There's some identical text between this and the -l option. For maintainability, could you please just leave one copy and refer to one from the other? For example, you might just say "see the description of the `link` option below" or something.

**Comment by kierdavis** [comment]
@sampsyo Thanks for the quick feedback. I did think it felt quite lengthy when I wrote it, but I wanted to ensure users are aware that embed (which is on by default) and link used together may cause the original files to be modified. Maybe a better solution would be to add an "are you sure?" prompt in the code when this combination of options is detected; what do you think?

**Comment by sampsyo** [comment]
Sure! Or here are two other ideas:

- Just a short phrase like "(don't use with `embed`, probably)" as a warning there.
- The plugin could just refuse to run with both flags enabled—I doubt anyone will actually want that behavior.

**Pull request closed without merging**